contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -638,11 +641,57 @@ is_numeric_datatype(Oid typid)
 	return decimal_oid == typid;
 }
 
+/*
+ * is_exact_numeric_datatype - returns bool if given datatype is numeric, decimal, int, tinyint, smallint, bigint, bit, money and smallmoney.
+ */
+static bool
+is_exact_numeric_datatype(Oid typid)
+{
+	typid = getBaseType(typid);
+
+	if (typid == NUMERICOID || typid == INT2OID || typid == INT4OID || typid == INT8OID)
+	{
+		return true;
+	}
+	if (!OidIsValid(decimal_oid))
+	{
+		TypeName *typename = makeTypeNameFromNameList(list_make2(makeString("sys"), makeString("decimal")));
+		decimal_oid = LookupTypeNameOid(NULL, typename, false);
+	}
+	if (!OidIsValid(tsql_tinyint_oid))
+	{
+		TypeName *typename = makeTypeNameFromNameList(list_make2(makeString("sys"), makeString("tinyint")));
+		tsql_tinyint_oid = LookupTypeNameOid(NULL, typename, false);
+	}
+	if (!OidIsValid(tsql_money_oid))
+	{
+		TypeName *typename = makeTypeNameFromNameList(list_make2(makeString("sys"), makeString("money")));
+		tsql_money_oid = LookupTypeNameOid(NULL, typename, false);
+	}
+	if (!OidIsValid(tsql_smallmoney_oid))
+	{
+		TypeName *typename = makeTypeNameFromNameList(list_make2(makeString("sys"), makeString("smallmoney")));
+		tsql_smallmoney_oid = LookupTypeNameOid(NULL, typename, false);
+	}
+	if (!OidIsValid(tsql_bit_oid))
+	{
+		TypeName *typename = makeTypeNameFromNameList(list_make2(makeString("sys"), makeString("bit")));
+		tsql_bit_oid = LookupTypeNameOid(NULL, typename, false);
+	}
+	
+	return (decimal_oid == typid
+			|| tsql_tinyint_oid == typid
+			|| tsql_money_oid == typid
+			|| tsql_smallmoney_oid == typid
+			|| tsql_bit_oid == typid);
```

Comment:
Seems inefficient to load all oids first and then return. Better if we decimal_oid == typid check after !OidIsValid(decimal_oid)

Reply:
Updated code